docs/local-development-environment.adoc

```diff
@@ -0,0 +1,182 @@
+
+Vagrant needs a base image to start off its VM. This base image is in
+its own format, called _box_. It's a tar-ball of a virtual disk and
+some vagrant-specific metadata. The virtual disk could be in one of
+several supported formats, depending on the virtualization provider to
+be used. The devbox will be using `qemu`, which is supported through
+Vagrant's `libvirt` provider.
+
+Vagrant usually hosts ready-to-use base-boxes of most OSes but there
+are no ready-made SLES4SAP images available. SUSE only provides SLES
+KVM/Xen images, mainly for use in OpenStack. Thus, for creating
+appropriate Vagrant `box` the following steps should be conducted:
+
+- Convert SLES KVM/Xen image to a box format.
+- Specialize the SLES base-box into a SLES4SAP VM appliance during first boot.
+
+=== SLES Vagrant base-box
+
+To create a SLES base-box, do the following steps:
+
+[NOTE]
+====
+These steps need to be executed only once for a SLES image of specific
+version. Eg. once for SLES15-SP6 and again when trying with SLES15-SP7.
+====
+
+. Download a desired SLES KVM/Xen image from `download.suse.com` or
+from https://scc.suse.com[SUSE Customer Center] (you'd need to have a
+customer account for the later). Again, there is no SLES4SAP VM images
+already prepared from SUSE. The good thing, though, is that both
+systems share the same base installation. Example name for the image
+could be `SLES15-SP6-Minimal-VM.x86_64-kvm-and-xen-GM.qcow2`.
+
+. Once you have the desired SLES image give its path to the provided
+`make_box.sh` shell script in `devbox` directory:
++
+[source,bash]
+----
+$ ./devbox/make_sles_box.sh <path-to-sles-image.qcow2>
+----
+
+
+WARNING: Ignore permission warnings. They are caused by when `fuser`
+syscall is not running as root. In most cases, these warnings are
+harmless.
+
+. After the script finishes, a new `*.box`
+file is created in `devbox` directory with a name extracted from the
+name of the original image. You have to add that box file to
+Vagrant. For instance, if the name of the produced file is
+`SLES15-SP6.box`, then:
++
+[source,bash]
+----
+$ vagrant box add --name SLES15-SP6 ./devbox/SLES15-SP6.box
+----
+
+Now, Vagrant recognizes `SLES15-SP6` as valid base box.
+
+=== SLES4SAP virtual appliance
+
+At this point you can start Vagrant using "base" SLES but the VM would
+require _interactive_ initial setup and also will not be registered
+with SCC. Most importantly, it will not be SLES4SAP but rather base
+SLES.
+
+To handle all of that, a Combustion script would be used on first boot
+to "specialize" the base-box into a virtual appliance ready to be used
+as a test target for the Ansible playbook.
+
+Because the real Combustion file should contain some secrets, the
+`devbox` directory contains a script called
+`make_combustion_script.sh` that would generate the real startup
+script with the secrets baked in:
+
+[NOTE]
+====
+The following procedure should be executed only once if secrets remain
```

Comment:
Try to avoid should be --> use active voice

Maybe something like 
Perform the following procedure only once if secrets remain the same, _regardless_ of the SLES version image in use: